File: land-backend/src/main/java/com/landmgmt/landbackend/model/Plot.java
package com.landmgmt.landbackend.model;

import java.math.BigDecimal;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Plot {

    private Long plotId;
    private String gaataNo;
    private Long landId;
    private String plotNo;
    private BigDecimal landLength;
    private BigDecimal landWidth;
    private BigDecimal sellRate;
    private BigDecimal totalAmount;
    private String status;

    private Long sellerId;
    private Seller seller;

    private Long brokerId;
    private Broker broker;
}

File: land-frontend/src/app/models/plot.ts
import { Seller } from './seller';
import { Broker } from './broker';


export interface Plot {
  plotId?: number;
  gaataNo: string;
  landId: number;
  plotNo: string;
  landLength: number;
  landWidth: number;
  sellRate: number;
  totalAmount: number;
  status: 'AVAILABLE' | 'SOLD' | 'RESERVED';
  sellerId?: number;
  seller?: Seller | null;
  brokerId?: number;
  broker?: Broker | null;
}